lib/Service/ApiService.php

```diff
@@ -210,6 +215,27 @@ public function push(Session $session, Document $document, $version, $steps, $aw
 		return new DataResponse($result);
 	}
 
+	private function addToPushQueue(Session $session, Document $document, int $version): void {
+		try {
+			$queue = Server::get(IQueue::class);
+			$syncResponse = $this->sync($session, $document, $version);
```

Comment:
I would expect to just add the steps that just arrived. Say we start at version 0 and two users push their steps without syncing in between:
* user1 pushes `step1`
* user2 pushes `step2`

Now the first push will add `[step1]` and the second push will add `[step1, step2]` to the queue because user2 still has version 0.

Or in other words.. the content of the queue gets send to all users and what is added to it should not depend on the version of the user adding it.

Reply:
y-websocket has a timer that will send Sync messages of type 1 periodically:
https://github.com/yjs/y-websocket/blob/35be3431b8db14f07dfaacab5dfa1a4cdec13fac/src/y-websocket.js#L300-L311

We could enable that. But if the frequency is too high it will create quite a bit of load as all clients would query all other clients. If the frequency is too low it will take too long to resync to be helpful.

My favorite approach right now would be for the saving client to also save a Sync message type 1 alongside the current state. Then every client can get that as a baseline and apply the saved state and reply to the message type 1 with a message type 2 that will contain its local diff.

Reply:
This should now be resolved by just using the notify_push queue as a fast path to distribute any incoming messages to other and the own clients.